rotkehlchen/chain/ethereum/uniswap/uniswap.py

```diff
@@ -0,0 +1,365 @@
+                param_types=param_types,
+                param_values=param_values,
+            )
+            result_data = result['tokenDayDatas']
+
+            if not result_data:
+                break
+
+            for tdd in result_data:
+                asset_price[to_checksum_address(tdd['token']['id'])] = FVal(tdd['priceUSD'])
+
+            # Update pagination step
+            param_values['offset'] += limit  # type: ignore
+
+        return asset_price
+
+    def _update_assets_prices_in_address_balances(
+            self,
+            address_balances: AddressBalances,
+            known_asset_price: AssetPrice,
+            unknown_asset_price: AssetPrice,
+    ) -> None:
+        """Update the pools underlying assets prices in USD (prices obtained
+        via Inquirer and the Uniswap subgraph)
+        """
+        for lps in address_balances.values():
+            for lp in lps:
+                # Try to get price from either known or unknown asset price.
+                # Otherwise keep existing price (zero)
+                total_user_balance = FVal(0)
+                for asset in lp.assets:
+                    asset_ethereum_address = asset.asset.ethereum_address
+                    asset_usd = known_asset_price.get(
+                        asset_ethereum_address,
+                        unknown_asset_price.get(asset_ethereum_address, ZERO),
+                    )
+                    # Update <LiquidityPoolAsset> if asset USD price exists
+                    if asset_usd != ZERO:
+                        asset.asset_usd = Price(asset_usd)
+                        asset.user_balance.usd_value = (
+                            asset.user_balance.amount * asset_usd
+                        )
+
+                    total_user_balance += asset.user_balance.usd_value
+
+                # Update <LiquidityPool> total balance in USD
+                lp.user_balance.usd_value = total_user_balance
+
+    def get_balances(
+        self,
+        addresses: List[ChecksumEthAddress],
+        graph_query: bool,
+    ) -> AddressBalances:
+        """Get the addresses' balances in the Uniswap protocol
+
+        Premium users can request balances either via the Uniswap subgraph or
+        Zerion SDK.
+        """
+        is_graph_mode = self.graph and self.premium and graph_query
+
+        if is_graph_mode:
+            protocol_balance = self._get_balances_graph(addresses)
+        else:
+            protocol_balance = self._get_balances_zerion(addresses)
+
+        known_assets = protocol_balance.known_assets
+        unknown_assets = protocol_balance.unknown_assets
+
+        known_asset_price = self._get_known_asset_price(known_assets, unknown_assets)
+
+        unknown_asset_price: AssetPrice = {}
+        if is_graph_mode:
+            unknown_asset_price = self._get_unknown_asset_price_graph(unknown_assets)
+
+        self._update_assets_prices_in_address_balances(
+            protocol_balance.address_balances,
```

Comment:
can you use kwargs here? We generally try to use kwargs when invoking a function call if it has more than 1-2 args as it's much easier to see what arg is what and also we guard against errors where you change the function's argument order and that results in exceptions. The interpret would see it and warn you if you use kwargs